packages/app/src/app/store/sequences.js

```diff
@@ -146,15 +147,25 @@ export const forkSandbox = sequence('forkSandbox', [
   },
 ]);
 
+export const forkFrozenSandbox = sequence('forkFrozenSandbox', [
+  when(state`editor.currentSandbox.isFrozen`),
+  {
+    true: [alertForkingFrozenSandbox, forkSandbox],
```

Comment:
To make this work with a `confirm` we can probably use the `path` directive of Cerebral. Here is an example: https://github.com/CompuIves/codesandbox-client/blob/master/packages/app/src/app/store/actions.js#L8-L19

The idea is that you can return different paths from an action, and in the sequence you can define what to do for every returned path. So here you could do:

```
{
	confirm: [forkSandbox],
    cancel: []
}
```

I think that would solve your question for how to use `confirm`.

Reply:
I've given this a try locally, and the fork is cancelled as we'd expect when user cancels the confirm 🎉

However the issue then becomes cancelling the remaining actions e.g:
![image](https://user-images.githubusercontent.com/21317379/47921404-70064380-deac-11e8-81ea-040a483b994e.png)

Here when `ensureOwnedEditable` returns without forking, the actions below are still dispatched (e.g. save is performed on frozen sandbox).

To be honest I feel as though it'd be too disruptive to the architecture to add in `if` `else` type logic, especially in this PR 😆

As this PR is a UX enhancement, i'll leave it as-is, feel free to reject/approve as you see fit 👍